model/registry/vcs.yaml

```diff
@@ -0,0 +1,65 @@
+groups:
+  - id: registry.vcs.repository
+    prefix: vcs.repository
+    type: attribute_group
+    brief: >
+     This group defines the attributes for
+     [Version Control Systems (VCS)](https://en.wikipedia.org/wiki/Version_control).
+    attributes:
+      - id: url
+        type: string
+        stability: experimental
+        brief: >
+          The [URL](https://en.wikipedia.org/wiki/URL) of the repository
+          providing the complete address in order to locate and identify the
+          repository.
+        examples: ['https://github.com/opentelemetry/open-telemetry-collector-contrib', 'https://gitlab.com/my-org/my-project/my-projects-project/repo']
+      - id: ref.name
+        type: string
+        stability: experimental
+        brief: >
+          The name of the [reference](https://git-scm.com/docs/gitglossary#def_ref) in the repository.
+        examples: ['my-feature-branch', 'tag-1-test']
+      - id: ref.type
+        type:
+          allow_custom_values: true
+          members:
+            - id: branch
+              value: branch
+              brief: "[branch](https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefbranchabranch)"
+              stability: experimental
+            - id: tag
+              value: tag
+              brief: "[tag](https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddeftagatag)"
+              stability: experimental
+        stability: experimental
+        brief: >
+          The type of the [reference](https://git-scm.com/docs/gitglossary#def_ref) in the repository.
+        examples: ['branch', 'tag']
+      - id: ref.commit.hash
+        type: string
+        stability: experimental
+        brief: >
+          The [hash value (see glossary)](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-5.pdf),
```

Comment:
I was thinking that distinguishing between short/full wouldn't be necessary in this attribute since it's simply the value and the value can be w/e comes from whatever tool you're using and how however you have it configured. If it's necessary to distinguish, we could do one or both of the following. 


`hash.value.{full, short}` or `hash.value.type = {sha1, ecda, etc}`

I'm personally partial to letting the hash value be non-distinguishable. But I'm also open to suggestion.

Reply:
I'm not suggesting to add attributes to record both, just to clarify that this one is the long one.

If it's not clarified than different tooling/instrumentation may end up adding different commit hashes and it'd be hard to query data generated by these tooling.

If we clarify that this attribute represents the full globally unique identifier, it'd be less likely.

Reply:
Changing it to read `The full [hash value....]`.